elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -666,9 +667,16 @@ public Boolean isIdGenerated() {
      * @return PersistentResource relation
      */
     public PersistentResource getRelation(String relation, String id) {
-
-        Predicate idFilter = new Predicate(relation, Operator.IN, Collections.singletonList(id));
-        Set<Predicate> filters = Collections.singleton(idFilter);
+        Set<Predicate> filters;
+        // TODO: UUID's are troublesome here from Patch Extension. We should consider an alternative approach for this.
+        if (requestScope instanceof PatchRequestScope) {
+            filters = Collections.emptySet();
+        } else {
+            Object idVal = CoerceUtil.coerce(id,
+                    dictionary.getIdType(dictionary.getParameterizedType(getResourceClass(), relation)));
+            Predicate idFilter = new Predicate("id", Operator.IN, Collections.singletonList(idVal));
```

Comment:
We could achieve this with `dictionary.getIdField` which could just return `entityBinding(value.getClass()).getIdField()`